src/main/java/donnafin/logic/PersonAdapter.java

```diff
@@ -71,7 +72,12 @@ public void edit(Name newName) {
                 curr.getAssets()
         );
         this.subject = personToEdit;
-        model.setPerson(curr, personToEdit);
+        try {
+            model.setPerson(curr, personToEdit);
+            model.saveAddressBook();
+        } catch (IOException e) {
+            throw new RuntimeException(e.getMessage());
+        }
```

Comment:
```suggestion
        trySavingAddressBook();
```

:warning: Do not commit this directly, this is not complete.

What if we replace all these duplicated code fragments with a call to a method that does exactly this. A private method `trySavingAddressBook()` in `PersonAdapter` would work right?